zebra-consensus/src/chain/tests.rs

```diff
@@ -133,42 +215,11 @@ async fn verify_block() -> Result<(), Report> {
 
     let (mut chain_verifier, _) = verifiers_from_checkpoint_list(Mainnet, checkpoint_list);
 
-    /// SPANDOC: Make sure the verifier service is ready for block 0
-    let ready_verifier_service = chain_verifier.ready_and().await.map_err(|e| eyre!(e))?;
-    /// SPANDOC: Set up the future for block 0
-    let verify_future = timeout(
-        Duration::from_secs(VERIFY_TIMEOUT_SECONDS),
-        ready_verifier_service.call(block0.clone()),
-    );
-    /// SPANDOC: Verify block 0
-    // TODO(teor || jlusby): check error kind
-    let verify_response = verify_future
-        .map_err(|e| eyre!(e))
-        .await
-        .expect("timeout should not happen")
-        .expect("block should verify");
-
-    assert_eq!(verify_response, hash0);
-
-    let block1 = Arc::<Block>::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_1_BYTES[..])?;
-    let hash1: BlockHeaderHash = block1.as_ref().into();
-
-    /// SPANDOC: Make sure the verifier service is ready for block 1
+    /// SPANDOC: Make sure the verifier service is ready
     let ready_verifier_service = chain_verifier.ready_and().await.map_err(|e| eyre!(e))?;
```

Comment:
surprised we need to ready up before running the transcript.

Reply:
for the transcript service specifically, poll_ready is a no-op, but it's required by the service contract.

Reply:
But this is isn't the transcript service we're calling `ready` on, this is the service that we're testing with the transcript. Checking the `Transcript::check` fn shows that the first thing it does is call `ready_and`, so this looks like a violation of the service contract to me, which is why I was surprised.

Reply:
Oh that is a good point, I think you're right.

Reply:
I have looked at the other transcript-based tests, and they do not call ready before checking.

Reply:
Removed all ready calls at https://github.com/ZcashFoundation/zebra/pull/858/commits/73ebc537a99d4b51201bcc600263c9e8b6cc905b